apps/dashboard/app/models/batch_connect/app.rb

```diff
@@ -215,6 +215,8 @@ def session_view
     # @return [String, nil] session info
     def session_info_view
       Pathname.new(root).glob("info.{md,html}.erb").find(&:file?).try(:read)
```

Comment:
memoize this

```
def session_info_view
  @session_info_view ||= Pathname.new(root).glob("info.{md,html}.erb").find(&:file?).try(:read)
...
```

otherwise when you call `session_info_view` to see if it exists and then call it again to get the contents of the view you have read the file contents twice